packages/safety-suite-api/src/api/models/types.ts

```diff
@@ -944,3 +944,197 @@ export type LostRestrictedInput = {
+   */
+  name: Alias;
+  /**
+   * Formatted display label.
+   */
+  label?: string;
+  /**
+   * The data type.
+   */
+  type: Type;
+}
+
+export interface BaseFieldOptions<Value extends unknown> {
+  /**
+   * How the value should be formatted. E.g. `currency` for a number type.
+   */
+  formatType?: string;
+  /**
+   * Default value if no value is provided upon creation of the model.
+   */
+  defaultValue?: Value;
+  /**
+   * If the field is included in custom reporting. Can be a boolean or an object to be checked against config.
+   *
+   * Defaults to `true`
+   */
+  onReport?: boolean | {key: string; values: unknown[]};
+  onColumn?: boolean;
+  /**
+   * Indirect modules which use this field.
+   */
+  indirectModules?: Alias[];
+}
+
+export type Required =
+  | boolean
+  | {
+      path: string[];
+      cases: {
+        pattern: string;
+        result: boolean;
+      }[];
+      default?: boolean;
+    };
+
+export type DateType = 'date' | 'time' | 'dateTime';
+
+export interface Field<Type extends FieldType = FieldType>
+  extends BaseField<Type> {
+  options: BaseFieldOptions<FieldTypes[Type]> & {
+    /**
+     * If the field is confidential.
+     */
+    protected: boolean;
+    /**
+     * If the field is required, can be a boolean or an object to be checked against config.
+     */
+    required: Required;
+    sideEffect?: unknown;
+  } & (Type extends 'list'
+      ? {
+          itemType: {
+            name: string;
+            type: FieldType;
+          };
+        }
+      : Type extends 'number'
+      ? {
+          /**
+           * Type of number. E.g. `decimal`
+           */
+          numberType?: string;
+          /**
+           * Minimum allowed value.
+           */
+          min?: number;
+        }
+      : Type extends 'dateTime'
+      ? {
+          errorOnHoliday?: boolean;
+          errorOnWeekend?: boolean;
+          onlyFuture?: boolean;
+          /**
+           * Format string. E.g. `MM/DD/YYYY`
+           */
+          format?: string;
+          type?: DateType;
+        }
+      : Type extends 'enum'
+      ? {
+          /**
+           * ID of the enum associated with this field.
+           */
+          enumSetId: number;
+          /**
+           * If the field can accept multiple values.
+           */
+          multi?: boolean;
+        }
+      : // eslint-disable-next-line @typescript-eslint/ban-types
+        {});
+}
+
+export interface Computation<ResultType extends FieldType = FieldType>
+  extends BaseField<'computed'> {
+  options: BaseFieldOptions<FieldTypes[ResultType]> & {
+    operation: string;
+    /**
+     * The data type of the computation result.
+     */
+    resultType?: ResultType;
+    ignoreBlank?: boolean;
+    separator?: string;
+    sortWith?: string;
+    status?: string;
+    /**
+     * When the computation will be run. E.g. `read`
+     */
+    computeOn?: string;
+    condition?: string;
+    ifTrue?: unknown;
+    ifFalse?: unknown;
```

Comment:
A lot of these fields are conditional based on other fields on here.  Eg, `ifTrue` / `ifFalse` are required if `condition` is one of `>`, `>=`, etc, etc.  If you want, I can map a bunch of these out, but I don't think the FE ever cares about these fields so might not be a good use of time.  Thoughts?

Reply:
Same thought as the above comment, probably not worth it if we don't intend on using these on the FE.